### PR TITLE
Show Build success if no tests are executed

### PR DESCRIPTION
**Pull request by JohannesStoehr**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

During an exam some courses don't provide any tests for programming exercises. But the compilation is still tested.
The feedback "0 of 0 tests passed" is not a helpful message in an exam. This should have a better message.

### Description

If no tests are present, but the build succeeded, a new message is shown in programming exercises as a score, that shows that the build compiled successfully.

### Steps for Testing

1. New Feature
    1. Create a programming exercise that doesn't have any tests that run before the due date. (No tests at all lead to a build failure)
    2. Submit a compiling code as student
    3. See the new compilation message.
2. Old exercises: make sure that normal exercises with tests still work:
    - A submission with no tests passing, but some tests present
    - A submission with build failure
    - A submission with some and one with all passing tests
    - A submission with manual feedback should work as normal

### Screenshots

Bildschirmfoto 2021-04-27 um 16 40 54

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/result/result.component.ts [hunk 1]
```diff
@@ -317,6 +319,20 @@ export class ResultComponent implements OnInit, OnChanges {
...
+     * Checks if only compilation was tested. This is the case, when a successful result is present with 0 of 0 passed tests
+     *
+     */
+    getOnlyShowSuccessfulCompileStatus(): boolean {
+        return (
+            this.templateStatus !== ResultTemplateStatus.NO_RESULT &&
+            this.templateStatus !== ResultTemplateStatus.IS_BUILDING &&
+            this.result != undefined &&
+            this.result!.resultString != undefined &&
+            this.result!.resultString === '0 of 0 passed'
```
Comment (ivanchimeno) [inline src/main/webapp/app/exercises/shared/result/result.component.ts:332]:
```suggestion
            this.result?.resultString? === '0 of 0 passed'
```

**Review by ivanchimeno — APPROVED** [review]
Code looks good, I just added a small optional comment to simplify the code

**Review by FuchsDominik — APPROVED** [review]
Code looks good :)

**Review by KiraNickel — DISMISSED** [review]
Tested on ts3. Old and new functionality work as expected 👍🏼

**Review by NicolasUllmann — APPROVED** [review]
Tested on server 3. Worked as expected.

**Review by maximilianbergmann — APPROVED** [review]
Tested. Works

**Review by niclasschuemann — DISMISSED** [review]

**Review by iozbeyli — APPROVED** [review]

**Review by iozbeyli — DISMISSED** [review]
Code looks good.

**Review by nicolasruscher — DISMISSED** [review]

**Review by katjanakosic — DISMISSED** [review]
Tested on TS1. Works as expected :)

**Review by cremertim — APPROVED** [review]
works on ts3

File: src/main/webapp/i18n/en/editor.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
             "buildOutput": "&nbsp; Build Output",
             "building": "Building and testing...",
             "buildFailed": "Build Failed",
+            "buildSuccessful": "Build Successful, no Tests Executed",
```
Comment (FrankeLukas) [inline src/main/webapp/i18n/en/editor.json:37]: We should probably be consistent with capitalization here, so either `No` or perhaps
```suggestion
            "buildSuccessful": "Build Successful, no tests executed",
```
?

File: src/main/webapp/app/exercises/shared/result/result.component.html [hunk 3]
```diff
@@ -14,11 +14,19 @@
             ><!-- we need to make sure that result is not null, otherwise the compiler complains below -->
             <span [ngClass]="textColorClass" class="guided-tour">
                 <fa-icon [icon]="resultIconClass" size="lg"></fa-icon>
-                <span class="score"> &nbsp;<span *ngIf="!short" jhiTranslate="artemisApp.result.score">Score</span> {{ round(result.score) }} %, </span>
+                <span class="score">
+                    <span *ngIf="!short" jhiTranslate="artemisApp.result.score">Score</span>
+                    {{ round(result.score) }} %,
```
Comment (FrankeLukas) [inline src/main/webapp/app/exercises/shared/result/result.component.html:19]:
```suggestion
                    {{ round(result.score) }}%,
```

**Review by FrankeLukas — COMMENTED** [review]
Code changes lgtm otherwise.

**Review by KiraNickel — CHANGES_REQUESTED** [review]
Tested on ts3. Old functionality works ~~but for exercises without tests run before the due date, 0 of 0 tests passed with red X is shown again for compiling submissions. 
Also it seems that when you submit in the Online Editor, Building and Testing... is shown indefinitely unless you go back to the overview of the exercise. (Not sure if this is connected to this PR but it didn't happen for the last version of this PR I tested)~~

**Review by KiraNickel — DISMISSED** [review]
Tested on ts3. Works as expected.
(Previously there seems to have been something wrong with the deploy)

**Review by ivanchimeno — DISMISSED** [review]
Re-approving

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 1. Re-approving.

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Worked for me.

**Review by cremertim — DISMISSED** [review]
reaproving after another working test on ts3

**Pull request merged**